src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -29,154 +30,59 @@ public LineService(LineRepository lineRepository, StationService stationService)
         this.stationService = stationService;
     }
 
+    @Transactional
     public LineResponse saveLine(LineRequest request) {
         Station upStation = stationService.findById(request.getUpStationId());
         Station downStation = stationService.findById(request.getDownStationId());
-        Line persistLine = lineRepository.save(new Line(request.getName(), request.getColor(), upStation, downStation, request.getDistance()));
-        List<StationResponse> stations = getStations(persistLine).stream()
-                .map(it -> StationResponse.of(it))
-                .collect(Collectors.toList());
-        return LineResponse.of(persistLine, stations);
+        Line persistLine = lineRepository.save(Line.of(request.getName(), request.getColor(), upStation, downStation, Distance.from(request.getDistance())));
+
+        return LineResponse.of(persistLine);
     }
 
+    @Transactional(readOnly = true)
     public List<LineResponse> findLines() {
-        List<Line> persistLines = lineRepository.findAll();
-        return persistLines.stream()
-                .map(line -> {
-                    List<StationResponse> stations = getStations(line).stream()
-                            .map(it -> StationResponse.of(it))
-                            .collect(Collectors.toList());
-                    return LineResponse.of(line, stations);
-                })
-                .collect(Collectors.toList());
+        List<Line> lines = lineRepository.findAll();
+
+        return lines.stream()
```

Comment:
`stream` 을 통해 변환하는 로직을 `Response` 클래스에서 수행해보는건 어떨까요?

스타일따라 다른 부분이라, 변경하지 않으셔도 됩니다 :)

ex)
```suggestion
        return LineResponse.ofList(line);
```